overlord/hookstate/hookmgr.go

```diff
@@ -227,6 +227,9 @@ func (m *HookManager) doRunHook(task *state.Task, tomb *tomb.Tomb) error {
 	handlers := m.repository.generateHandlers(context)
 	handlersCount := len(handlers)
 	if handlersCount == 0 {
+		if hooksup.Optional {
```

Comment:
Can we please have a comment here on 2.28+?  For 2.27 this is fine, but this line is not really obvious by itself.